.github/workflows/wasm-size.yml

```diff
@@ -0,0 +1,111 @@
+name: Report wasm size
+on:
+  pull_request:
+    paths-ignore:
+      - '.github/**'
+      - '!.github/workflows/wasm-size.yml'
+      - '.buildkite/**'
+      - '*.md'
+      - 'LICENSE'
+      - 'CODEOWNERS'
+      - 'renovate.json'
+
+jobs:
+  pr-wasm-size:
+    name: Get PR module size
+    runs-on: ubuntu-latest
+    outputs:
+      size: ${{ steps.measure.outputs.size }} 
+      size_gz: ${{ steps.measure.outputs.size_gz }} 
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+      
+      - uses: dtolnay/rust-toolchain@stable
+
+      - name: 'Install wasm-pack'
+        run: cargo install wasm-pack
+      
+      - name: Build and measure PR branch
+        id: measure
+        run: |
+          make build-qe-wasm
+          echo "size=$(wc --bytes < query-engine/query-engine-wasm/pkg/query_engine_bg.wasm)" >> $GITHUB_OUTPUT
+          echo "size_gz=$(gzip -c query-engine/query-engine-wasm/pkg/query_engine_bg.wasm | wc -c)" >> $GITHUB_OUTPUT
+
+  base-wasm-size:
+    name: Get base branch size
+    runs-on: ubuntu-latest
+    outputs:
+      size: ${{ steps.measure.outputs.size }} 
+      size_gz: ${{ steps.measure.outputs.size_gz }} 
+    steps:
+      - name: Checkout base branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.base.sha }}
+      
+      - uses: dtolnay/rust-toolchain@stable
+
+      - name: 'Install wasm-pack'
+        run: cargo install wasm-pack
+      
+      - name: Build and measure base branch
+        id: measure
+        run: |
+          make build-qe-wasm
+          echo "size=$(wc -c < query-engine/query-engine-wasm/pkg/query_engine_bg.wasm)" >> $GITHUB_OUTPUT
+          echo "size_gz=$(gzip -c query-engine/query-engine-wasm/pkg/query_engine_bg.wasm | wc -c)" >> $GITHUB_OUTPUT
+  
+  report-diff:
+    name: Report module size difference
+    runs-on: ubuntu-latest
+    needs: 
+      - pr-wasm-size 
+      - base-wasm-size
+
+    steps:
+      - name: Compute difference
+        id: compute
+        run: |
+          base=$(echo "${{ needs.base-wasm-size.outputs.size }}" | numfmt --to=iec-i --suffix=B)
+          base_gz=$(echo "${{ needs.base-wasm-size.outputs.size_gz }}" | numfmt --to=iec-i --suffix=B)
+          pr=$(echo "${{ needs.pr-wasm-size.outputs.size }}" | numfmt --to=iec-i --suffix=B)
+          pr_gz=$(echo "${{ needs.pr-wasm-size.outputs.size_gz }}" | numfmt --to=iec-i --suffix=B)
+          
+          diff=$(echo "$((${{ needs.pr-wasm-size.outputs.size }} - ${{ needs.base-wasm-size.outputs.size }}))" | numfmt --to=iec-i --suffix=B)
+          diff_gz=$(echo "$((${{ needs.pr-wasm-size.outputs.size_gz }} - ${{ needs.base-wasm-size.outputs.size_gz }}))" | numfmt --to=iec-i --suffix=B)
```

Comment:
```suggestion
          base=$(echo "${{ steps.base-wasm-size.outputs.size }}" | numfmt --to=iec-i --suffix=B)
          base_gz=$(echo "${{ steps.base-wasm-size.outputs.size_gz }}" | numfmt --to=iec-i --suffix=B)
          pr=$(echo "${{ steps.pr-wasm-size.outputs.size }}" | numfmt --to=iec-i --suffix=B)
          pr_gz=$(echo "${{ steps.pr-wasm-size.outputs.size_gz }}" | numfmt --to=iec-i --suffix=B)
          
          diff=$(echo "$((${{ steps.pr-wasm-size.outputs.size }} - ${{ steps.base-wasm-size.outputs.size }}))" | numfmt --to=iec-i --suffix=B)
          diff_gz=$(echo "$((${{ steps.pr-wasm-size.outputs.size_gz }} - ${{ steps.base-wasm-size.outputs.size_gz }}))" | numfmt --to=iec-i --suffix=B)
```

Reply:
Do you know if what you wrote works, I think you would need to use `steps` here

Reply:
It looks like it's not needed, since it worked

Reply:
I don't think it work with `steps`: these are separate jobs that run in parallel.

Reply:
Right!